superset/assets/src/visualizations/core/models/ChartPlugin.js

```diff
@@ -0,0 +1,43 @@
+import Plugin from './Plugin';
+import isRequired from '../../../utils/isRequired';
+import { getInstance as getChartMetadataRegistry } from '../registries/ChartMetadataRegistry';
+import { getInstance as getChartComponentRegistry } from '../registries/ChartComponentLoaderRegistry';
+import { getInstance as getChartTransformPropsRegistry } from '../registries/ChartTransformPropsLoaderRegistry';
+
+const IDENTITY = x => x;
+
+export default class ChartPlugin extends Plugin {
+  constructor({
+    metadata = isRequired('metadata'),
+
+    // use transformProps for immediate value
+    transformProps = IDENTITY,
+    // use loadTransformProps for dynamic import (lazy-loading)
+    loadTransformProps,
+
+    // use Chart for immediate value
```

Comment:
can you explain a little more when you would want to use `Chart` over `loadChart`, or 'transformProps` over `loadTransformProps`? especially since under the hood it seems that they are all transformed to loader `func`s anyway?

Reply:
If the chart is very small. Use `Chart`, this will include `Chart` in the bundle instead of spinning it of into a lazy-loaded bundle.

```
import Chart from './Chart';
...
chart: Chart,
...
```

if the chart is gigantic. Use `loadChart`. 

```
loadChart: () => import('./Chart.jsx')
```